app/addons/setup/container/MultipleNodeContainer.js

```diff
@@ -0,0 +1,29 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+import {connect} from 'react-redux';
+import MultipleNodeController from '../components/MultipleNodeController';
+import {getNodeList, getIsAdminParty, getAdditionalNode, getUsername, getPassword, getSetupNode} from '../reducers';
+
+const mapStateToProps = ({setup}) => {
+  return {
+    nodeList: getNodeList(setup),
+    isAdminParty: getIsAdminParty(setup),
+    setupNode: getSetupNode(setup),
+    username: getUsername(setup),
+    password: getPassword(setup),
+    additionalNode: getAdditionalNode(setup)
+  };
+};
+
```

Comment:
With all your controllers you are using doing `this.dispatch(theAction);`. That isn't correct. Could you rather create another function here called `mapDispatchToProps` can create the required actions for that controller here. See https://github.com/apache/couchdb-fauxton/blob/master/app/addons/documents/rev-browser/container.js#L55-L77 for an example.

Reply:
The container files are the only place where you should have to `import` the actions.